File: adventofcode/java/year2016/day22/Day22Test.java
package adventofcode.java.year2016.day22;

import java.io.FileReader;
import java.io.IOException;

import org.junit.Test;
import org.junit.Assert;

public class Day22Test {
    @Test
    public void part1Actual() throws IOException {
        try (var input = new FileReader("adventofcode/data/year2016/day22/actual.in")) {
            var output = "888";
            var result = Day22.part1(input);
            Assert.assertEquals("no error", "", result.error);
            Assert.assertEquals("correct output", output, result.answer);
        }
    }

     @Test
     public void part2Example() throws IOException {
        try (var input = new FileReader("adventofcode/java/testdata/year2016/day22/example")) {
            var output = "7";
            var result = Day22.part1(input);
            Assert.assertEquals("no error", "", result.error);
            Assert.assertEquals("correct output", output, result.answer);
        }
     }

     @Test
     public void part2Actual() throws IOException {
         try (var input = new FileReader("adventofcode/data/year2016/day22/actual.in")) {
             var output = "236";
             var result = Day22.part2(input);
             Assert.assertEquals("no error", "", result.error);
             Assert.assertEquals("correct output", output, result.answer);
         }
     }
}

File: adventofcode/cpp/year2019/day22/day22_test.cc
#include "adventofcode/cpp/year2019/day22/day22.h"

#include <fstream>
#include <string>

#include "gtest/gtest.h"

#include "adventofcode/cpp/adventofcode.h"

TEST(Year2019Day22, Part1Actual) {
  std::ifstream input("adventofcode/data/year2019/day22/actual.in");
  std::string output = "4485";
  adventofcode::answer_t a = day22::part1(input);
  EXPECT_EQ("", a.error);
  EXPECT_EQ(output, a.answer);
  input.close();
}

TEST(Year2019Day22, Part2Actual) {
  std::ifstream input("adventofcode/data/year2019/day22/actual.in");
  std::string output = "91967327971097";
  adventofcode::answer_t a = day22::part2(input);
  EXPECT_EQ("", a.error);
  EXPECT_EQ(output, a.answer);
  input.close();
}
